src/yanfeishao/cs555/utils/DateUtils.java

```diff
@@ -186,8 +182,14 @@ public void parseUS03Error(Set<String> result, String prefix, FamilyEntity famil
      *         the divorce date
      */
     public void parseUS04Error(Set<String> result, String prefix, FamilyEntity familyEntity, Date marriageDate, Date divorceDate) {
-        if (us04ParseCondition(marriageDate, divorceDate)) {
-            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US04_FAMILY, prefix, familyEntity.getIdentifier()));
+        if (CommonUtils.isNotNull(marriageDate) && CommonUtils.isNotNull(divorceDate)) {
+            if (prefix.equals(ErrorCode.US04)) {
+                if (marriageDate.after(divorceDate)) {
```

Comment:
Change this to if (marriageDate.after(divorceDate) || divorceDate.before(marriageDate)){
...
}